inginious/agent/docker_agent/_docker_interface.py

```diff
@@ -61,16 +71,15 @@ def get_host_ip(self, env_with_dig='ingi/inginious-c-default'):
             return None
 
     def create_container(self, environment, network_grading, mem_limit, task_path, sockets_path,
-                         course_common_path, course_common_student_path, ssh_port=None):
+                         course_common_path, course_common_student_path, ports={}):
```

Comment:
This is a bad smell in Python. Set `ports=None` and then in the code below add the lines `if ports is None: ports = {}`. 

If you don't see why it's bad to do it like that, check in the python doc ;-)